### PR TITLE
Adding Kakoune

### PR DESCRIPTION
**Pull request by eunos-1128**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by eunos-1128** [comment]
@conda-forge/help-c-cpp

**Comment by eunos-1128** [comment]
@conda-forge/help-c-cpp

**Comment by eunos-1128** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/kakoune/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -exo pipefail
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
```
Comment (xhochy) [inline recipes/kakoune/build.sh:5]:
```suggestion
if [[ "${target_platform}" == linux-* ]]; then
```

File: recipes/kakoune/build.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    make install -j${CPU_COUNT} \
+        debug=no \
+        PREFIX=${PREFIX} \
+        CXX=${CXX}
+
+elif [[ "$OSTYPE" == "darwin"* ]]; then
```
Comment (xhochy) [inline recipes/kakoune/build.sh:11]:
```suggestion
elif [[ "${target_platform}" == "darwin"* ]]; then
```

File: recipes/kakoune/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - {{ stdlib("c") }}
+    - llvm  # [osx]
+    - make
+    - binutils  # [linux]
+  host:
+    - libcxx  # [osx]
+  run:
+    - libcxx  # [osx]
```
Comment (xhochy) [inline recipes/kakoune/meta.yaml:28]: They should come in automatically using the compiler
```suggestion
```

File: recipes/kakoune/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - {{ stdlib("c") }}
+    - llvm  # [osx]
```
Comment (xhochy) [inline recipes/kakoune/meta.yaml:22]: This should come in via the compiler or does it explicitly use LLVM features?
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/kakoune/meta.yaml (hunk 4 above)
Comment (eunos-1128) [reply recipes/kakoune/meta.yaml]: > This should come in via the compiler 
As you say it does.

> does it explicitly use LLVM features?
Stuff coming with the compiler seems enough for the package to work.

File: recipes/kakoune/build.sh (hunk 2 above)
Comment (eunos-1128) [reply recipes/kakoune/build.sh]: For MacOS, `target_platform` value is not `darwin` but `osx-*`.

**Review by xhochy — APPROVED** [review]

**Pull request merged**